gnuradio-runtime/include/gnuradio/logger.h

```diff
@@ -156,136 +156,158 @@ typedef log4cpp::Category* logger_ptr;
+        if (logger->isPriorityEnabled(log4cpp::Priority::CRIT))          \
+            *logger << log4cpp::Priority::CRIT << (msg) << log4cpp::eol; \
     }
 
-#define GR_LOG_EMERG(logger, msg)                                     \
-    {                                                                 \
-        *logger << log4cpp::Priority::EMERG << (msg) << log4cpp::eol; \
+#define GR_LOG_ALERT(logger, msg)                                         \
+    {                                                                     \
+        if (logger->isPriorityEnabled(log4cpp::Priority::ALERT))          \
+            *logger << log4cpp::Priority::ALERT << (msg) << log4cpp::eol; \
     }
 
-#define GR_LOG_ERRORIF(logger, cond, msg)                                 \
+#define GR_LOG_FATAL(logger, msg)                                         \
     {                                                                     \
-        if ((cond)) {                                                     \
-            *logger << log4cpp::Priority::ERROR << (msg) << log4cpp::eol; \
-        }                                                                 \
+        if (logger->isPriorityEnabled(log4cpp::Priority::FATAL))          \
+            *logger << log4cpp::Priority::FATAL << (msg) << log4cpp::eol; \
     }
 
-#define GR_LOG_ASSERT(logger, cond, msg)                                  \
+#define GR_LOG_EMERG(logger, msg)                                         \
     {                                                                     \
-        if (!(cond)) {                                                    \
+        if (logger->isPriorityEnabled(log4cpp::Priority::EMERG))          \
             *logger << log4cpp::Priority::EMERG << (msg) << log4cpp::eol; \
-            assert(0);                                                    \
-        }                                                                 \
+    }
+
+#define GR_LOG_ERRORIF(logger, cond, msg)                                     \
+    {                                                                         \
+        if ((cond)) {                                                         \
+            if (logger->isPriorityEnabled(log4cpp::Priority::ERROR))          \
+                *logger << log4cpp::Priority::ERROR << (msg) << log4cpp::eol; \
+        }                                                                     \
+    }
+
+#define GR_LOG_ASSERT(logger, cond, msg)                                      \
+    {                                                                         \
+        if (!(cond)) {                                                        \
+            if (logger->isPriorityEnabled(log4cpp::Priority::EMERG))          \
+                *logger << log4cpp::Priority::EMERG << (msg) << log4cpp::eol; \
+            assert(0);                                                        \
```

Comment:
A possible improvement would be to add message to assert so the logged message is shown on command line as well. That would require additional macro. Something like 
#ifdef NDEBUG
# define assertm(exp, msg)  ((void)(0))
#else
# define assertm(exp, msg)                      \
    if (!exp) {                                              \
         std::cerr << "Assertion: " << msg;   \
         assert(exp);                                      \
     }                                  
#endif